modules/keymaps.nix

```diff
@@ -48,7 +71,30 @@ with lib; {
             vim.keymap.set(map.mode, map.key, map.action, map.options)
           end
         end
-        -- }}}
-      '';
+      ''
+      + foldl (a: b: a + "\n" + b)
+      ""
+      (
+        mapAttrsToList
```

Comment:
We have a wrapper for auto commands. I think you should be able to use it instead of calling directly vim.api.nvim_create_autocmd

Reply:
Sorry if I'm being an idiot here but I don't know how to use the helpers in `lib/autocmd-helpers.nix`. Looks to me just options and types and things, not any machinery for actually defining autocmds.
Additionally, `modules/keymaps.nix` evaluates to just a string, so I don't know how to also set nixvim configs there as well.

Reply:
You would be able to add in the `config` section the following:
```nix
   autoCmd = mapAttrsToList (
       event: mappings: {
            inherit event;
            group = helpers.mkRaw ''vim.api.nvim_create_augroup("nixvim_binds_${event}", { clear = true })'';
            callback = helpers.mkRaw ''
                 function()
                    do
                      local __nixvim_binds = ${helpers.toLuaObject (map normalizeMapping mappings)}
                      for i, map in ipairs(__nixvim_binds) do
                        vim.keymap.set(map.mode, map.key, map.action, map.options)
                      end
                    end
                 end
            '';
       }
   ) config.keymapsOnEvents;
```

Though maybe we need to allow `autoCmd.*.group` to be a raw lua string